tests/for_loop_test_suite/for_loop_test.cpp

```diff
@@ -0,0 +1,21 @@
+#include <gtest/gtest.h>
+#include <vector>
+#include <memory>
+#include "../../include/exceptions/user_error_tracker.h"
+#include "../../include/value.h"
+#include "../../include/visitors/interpreter.h"
+#include "../../src/callable.cpp"
+#include "../helpers/parse_test_helper.hpp"
+
+TEST(ForLoopTest, ForLoopIncrement)
+{
+    auto code = "for var x: int = 0; x <= 5; x += 1 do print x;";
+    auto ast = parse_code(code);
+
+    Interpreter interpreter;
+    interpreter.evaluate(&ast);
+
+    ASSERT_TRUE(interpreter.environment->contains("x"));
```

Comment:
x should not be available in the environment after the `for` loop exits. I believe this is an issue.